sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -258,9 +272,37 @@ static pi_result USMSetIndirectAccess(pi_kernel kernel) {
 
 extern "C" {
 
+// Return sub-device level
+// 0 -> root device
+// 1 -> sub-device
+// 2 -> sub-sub-device (CCS)
+// -1 -> invalid device
+static int getSubLevel(pi_device device) {
+  if (!device)
+    return -1;
+  cl_device_id parentId = nullptr;
+  clGetDeviceInfo(cast<cl_device_id>(device), CL_DEVICE_PARENT_DEVICE,
+                  sizeof(cl_device_id), &parentId, NULL);
+  if (parentId == nullptr)
+    return 0;
+  cl_device_id parentParentId = nullptr;
+  clGetDeviceInfo(parentId, CL_DEVICE_PARENT_DEVICE, sizeof(cl_device_id),
+                  &parentParentId, NULL);
+  if (parentParentId == nullptr)
+    return 1;
+  cl_device_id parentParentParentId = nullptr;
+  clGetDeviceInfo(parentParentId, CL_DEVICE_PARENT_DEVICE, sizeof(cl_device_id),
+                  &parentParentParentId, NULL);
+  if (parentParentParentId == nullptr)
+    return 2;
+  return -1;
```

Comment:
nit: I'd rather see a generic implementation (i.e., `while` loop) with an assert that the result is <=2.

Reply:
I changed the levels to use an enum. So, I am leaving code as is (no while loop). Hope that's ok. Thanks